spec/onchain.tex

```diff
@@ -315,9 +315,67 @@ \subsection{CollectCom Transaction}\label{sec:collect-tx}
 		$\stOpen$ head output.}\label{fig:collectComTx}
 \end{figure}
 
+\subsection{\red{Increment Transaction}}\label{sec:increment-tx}
+
+\noindent The \mtxIncrement{} transaction (Figure~\ref{fig:incrementTx}) allows a party to
+add a UTxO to an already open head and consists of:
+
+\begin{itemize}
+	\item one input spending from $\nuHead$ holding the $\st$ with $\datumHead$,
+	\item one output paying to $\nuHead$ with value $\valHead'$ and
+		  datum $\datumHead'$,
+    \item one or more inputs spent in this transaction $i_{1} \dots i_m$ where $m$ is the number of inputs to include and $\forall i \in \{i_{1} \dots i_{m}\}, i \notin I_{head}$.
+\end{itemize}
+
+\noindent The state-machine validator $\nuHead$ is spent with
+$\redeemerHead = (\mathsf{increment}, \xi)$, where $\xi$ is a multi-signature of
+the increment snapshot which authorizes addition of some UTxO $U_\alpha$. The
+validator checks:
+\begin{menumerate}
+	\item State is advanced from $\datumHead \sim \stOpen$ to
+	$\datumHead' \sim \stOpen$, parameters $\cid,\hydraKeysAgg,\nop,\cPer$
+	stay unchanged and the new state is governed again by $\nuHead$:
+	\[
+		(\stOpen,\cid,\hydraKeysAgg,\nop,\cPer,v,\eta) \xrightarrow[\xi]{\mathsf{increment}} (\stOpen,\cid,\hydraKeysAgg,\nop,\cPer,v',\eta')
+	\]
+	\item Increment snapshot number $s'$ is higher than the currently stored snapshot number $s$
+	\[
+		s' > s
+	\]
+	\item New version $v'$ is incremented correctly
+	\[
+		v' = v + 1
+	\]
+\item $\xi$ is a valid multi-signature of the new snapshot state (currency id $\cid$, the current state version $v$, the new state $\eta'$, UTxO to deposit $\eta_\alpha$ and UTxO to withdraw $\eta_\omega$)
+	\[
+    \msVfy(\hydraKeys,(\cid || v || s || \eta' || \eta_\alpha || \eta_\omega),\xi) = \true
+	\]
+  where snapshot number $s$ is provided through the redeemer, $\eta_\alpha$ is the digest of all added UTxO, and $\eta_\omega$ is the digest of all removed UTxO but in case of increment transaction $\eta_\omega = \bot$.
```

Comment:
Should: just put $\bot$ directly into the expression above. No need to explain `$\eta_\omega$ here.